### PR TITLE
Add controller and example for using on-board PD control on Solo12

### PR DESCRIPTION
**Pull request by luator**
## Description

- Add a `HoldOnBoardPDController` which is a copy of `HoldPDController` but using the on-board PD controller of the master board.
- Add an example `solo12_real_onboard-pd.py` which uses this controller.

## How I Tested

Tested on the 12-joint test bench.

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [x] https://github.com/open-dynamic-robot-initiative/solo/pull/111
- [x] https://github.com/open-dynamic-robot-initiative/robot_properties_solo/pull/48

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: src/dynamic_graph_head/controllers.py [hunk 1]
```diff
@@ -29,18 +30,119 @@ def __init__(self, head, Kp, Kd, with_sliders=False):
...
+            self.des_position = self.zero_pos
+
+        self.tau = (
+            self.Kp * (self.des_position - self.joint_positions)
+            - self.Kd * self.joint_velocities
+        )
+        self.head.set_control("ctrl_joint_torques", self.tau)
+
+
+class HoldOnBoardPDController:
```
Comment (jviereck) [inline src/dynamic_graph_head/controllers.py]: I wonder if you could make `HoldOnBoardPDController` a subclass from `HoldPDController`? This way the `map_sliders` etc. method would need to be implemented only once and reused across the two classes.

What do you think?

**Review by jviereck — CHANGES_REQUESTED** [review]

File: src/dynamic_graph_head/controllers.py (hunk 1 above)
Comment (luator) [reply src/dynamic_graph_head/controllers.py]: Good point

Comment (luator) [reply src/dynamic_graph_head/controllers.py]: Not yet tested, so please don't merge yet!

**Review by jviereck — APPROVED** [review]
This looks good.

**Comment by jviereck** [comment]
I approved the current version already. Once you tested it and it works, feel free to merge the PR.

**Pull request merged**